### PR TITLE
Fluent APIs are not allowed in non-builder types

### PR DESCRIPTION
**Pull request by adamsitnik**
Based on the feedback from @bartonjs

### CODE REVIEW

File: src/System.CommandLine.Tests/CompletionTests.cs [hunk 1]
```diff
@@ -968,5 +962,19 @@ public void When_option_completions_are_available_then_they_are_suggested_when_a
...
+            Argument<string> argument = new();
+            argument.AcceptOnlyFromAmong(values);
+            return argument;
+        }
+
+        private static Option<string> CreateOptionWithAcceptOnlyFromAmong(string name, params string[] values)
+        {
+            Option<string> argument = new(name);
+            argument.AcceptOnlyFromAmong(values);
+            return argument;
```
Comment (Jozkee) [inline src/System.CommandLine.Tests/CompletionTests.cs:977]: nit
```suggestion
            Option<string> option = new(name);
            option.AcceptOnlyFromAmong(values);
            return option;
```

File: src/System.CommandLine.Tests/OptionTests.cs [hunk 2]
```diff
@@ -372,10 +372,8 @@ public void Option_of_enum_can_limit_enum_members_as_valid_values()
         [Fact]
         public void Option_of_T_fluent_APIs_return_Option_of_T()
```
Comment (Jozkee) [inline src/System.CommandLine.Tests/OptionTests.cs:373]: You might as well remove the test

**Review by Jozkee — APPROVED** [review]
> Should the ArgumentValidation.AcceptExistingOnly and OptionValidation.AcceptExistingOnly methods likewise return void?

Up to Adam's choice, but that could be addressed in a follow-up PR if required.

**Comment by adamsitnik** [comment]
> Should the ArgumentValidation.AcceptExistingOnly and OptionValidation.AcceptExistingOnly methods likewise return void?

That is a very good question. To be honest I don't like `AcceptLegalFilePathsOnly` and `AcceptLegalFileNamesOnly` as they don't restrict the argument/option type in any way. Example:

```cs
Option<bool> flag = new ("--flag");
flag.AcceptLegalFilePathsOnly(); // it should work only with files, not things like boolean
```

`ArgumentValidation.AcceptExistingOnly` and `OptionValidation.AcceptExistingOnly ` restrict the types, but our BCL guidelines are against introducing extension methods for types that we own (they should be just instance methods). When @bartonjs and @jonsequitur are back from holidays I am going to discuss it with them.

**Pull request merged**